### DISCUSSION TITLE
Be aware that if you run XDEBUG and try to generate MPDF documents it will always fail

### DISCUSSION THREAD

**Discussion by stefan-ma22** [mpdf/mpdf#2009] [General]
I have found very strange behaviour when using XDEBUG and debugging is ON in Vscode. When I try to generate a pdf I always get this error **Error detected. PDF file generation aborted: Undefined variable $e**

When I turn off debugging in vscode strangely it works, I think it has something with xdebug session rewriting directory / file paths to when debugging session is on. I spent couple of hours on this looking for any errors so now I want to save other peoples time.

**Comment by finwe** [comment]
I'd be careful with the "always", code coverage for releases is run with xdebug and all test PDF generations end with no errors. https://github.com/mpdf/mpdf/actions/runs/8263060966/job/22603684101

**Reply by stefan-ma22** [reply]
Still I dont understand why when I run the xdebug session in vscode I get this error and when I turn off debugging it just works VSCode version 1.88.0-insider, WSL 2

xdebug config 
```
{
            "name": "Listen for Xdebug",
            "type": "php",
            "request": "launch",
            "port": 9003,
            "xdebugSettings": {
                "max_data": 65535,
            },
        },
```

**Reply by finwe** [reply]
Unless you provide a full reproducible example, no one can understand either.
